src/ccstruct/boxread.cpp

```diff
@@ -32,9 +33,26 @@
 // Special char code used to identify multi-blob labels.
 static const char* kMultiBlobLabelCode = "WordStr";
 
+// Returns the box file name corresponding to the given image_filename.
+static std::string BoxFileName(const char* image_filename) {
+  std::string box_filename = image_filename;
+  size_t length = box_filename.length();
+  std::string last = (length > 8) ? box_filename.substr(length - 8) : "";
+  if (last == ".bin.png" || last == ".nrm.png") {
+    box_filename.resize(length - 8);
+  } else {
+    const char* lastdot = strrchr(box_filename.c_str(), '.');
```

Comment:
Use str.find('.')?

Reply:
Good idea. `str.find_last_of('.')` might even be better. I'll update the PR tomorrow.

Reply:
Not sure about `find_last_of()`.